tools/pipelines/templates/build-npm-package.yml

```diff
@@ -357,19 +357,20 @@ extends:
                           testCoverage: ${{ parameters.testCoverage }}
 
                   - ${{ if contains(convertToJson(parameters.taskTest), 'tinylicious') }}:
-                    # Copy files so all artifacts we publish end up under the same parent folder.
-                    # Source is wherever the tinylicious log ends up in, after executing tests above.
-                    # CopyFiles@2 failed with OOM for some reason so doing it "by hand".
                     - task: Bash@3
-                      displayName: Copy tinylicious.log to artifact staging directory
+                      displayName: Upload tinylicious log
                       condition: succeededOrFailed()
                       continueOnError: true # Keep running subsequent tasks even if this one fails (e.g. the tinylicious log wasn't there)
                       inputs:
                         targetType: inline
-                        workingDirectory: '${{ parameters.buildDirectory }}/packages/test/test-end-to-end-tests'
                         script: |
-                          mkdir -p $(Build.ArtifactStagingDirectory)/tinyliciousLog
-                          cp tinylicious.log $(Build.ArtifactStagingDirectory)/tinyliciousLog/tinylicious.log
+                          PATH_TO_TINYLICIOUS_LOG=$(Build.SourcesDirectory)/packages/test/test-end-to-end-tests/tinylicious.log;
+                          if [ -f $PATH_TO_TINYLICIOUS_LOG ] ; then
+                            echo "Found file at '$PATH_TO_TINYLICIOUS_LOG'. Uploading.";
+                            echo "##vso[task.uploadfile]$PATH_TO_TINYLICIOUS_LOG";
```

Comment:
I tried reading up on it here but still confused on what vso does. Is it used to communicate that an task/action is taking place like upload file or log/print a warning? Is it mix between a log and a command?

https://learn.microsoft.com/en-us/azure/devops/pipelines/scripts/logging-commands?view=azure-devops&tabs=bash

Reply:
The way you invoke it is odd because you use the echo command but essentially these are some built in commands you can use in Azure pipelines. Azure pipelines have built in support for logs that you can download, and you can add additional log information to that downloadable set using this command If I am reading the docs correctly.

Reply:
It's more of a command to ADO. You're on the right place with the docs, there you'll see that we can use this kind of syntax to tell ADO things like "set this pipeline variable to this value", "print this message in the pipeline logs", "report that this step is 50% done", "upload this file to logs we can download', "add a tag to the pipeline run".